### ISSUE TITLE
unsolvable (typedef) error number 2

### ISSUE THREAD

**Issue by rogierschouten** [palantir/tslint#143]
The following code results in a (typedef) error and it should not. TypeScript does not allow to specify the return type for a lambda function. Changing to function() syntax is not a workaround because then I need to define "this" myself again.

``` javascript
    this._disposeImpl(() => {
        // do work
    });
```

``` shell
(typedef) disposable.ts[72, 26]: expected callSignature to have a typedef.
```
